antispam_bee.php

```diff
@@ -2359,12 +2361,17 @@ public static function get_client_ip() {
 			$ip = wp_unslash( $_SERVER['HTTP_FORWARDED_FOR'] );
 		} elseif ( isset( $_SERVER['HTTP_FORWARDED'] ) ) {
 			$ip = wp_unslash( $_SERVER['HTTP_FORWARDED'] );
-		} elseif ( isset( $_SERVER['REMOTE_ADDR'] ) ) {
+		}
+
+		preg_match_all ('/(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})/g', $ip, $matches );
```

Comment:
does not match ipv6,
`Unknown modifier 'g'`

Reply:
what about `filter_var($ip, FILTER_VALIDATE_IP)`?

Reply:
Also does not match short notations in v4, but it matches invalid records like ### _345.678.987.654_.

Either split, trim and filter separately, e.g. using PHP's built-in filter or use a better RegEx.

Also care about delimiters when using global mode patterns.  _:ffff:192.0.2.1_ is a valid address (IPv6 mapped IPv4) and must not be split in two.